packages/sdk/src/models/issue.ts

```diff
@@ -0,0 +1,150 @@
+import { gql } from "gql";
+import { Linear } from "../linear";
+import {
+  CreateIssueFromCliMutation,
+  CreateIssueFromCliMutationVariables,
+  UpdateIssueFromCliMutation,
+  UpdateIssueFromCliMutationVariables,
+  GetTeamIssueIdsFromCliQuery,
+  GetTeamIssueIdsFromCliQueryVariables,
+  GetIssueFromCliQuery,
+  GetIssueFromCliQueryVariables,
+} from "./issue.generated";
+import { getTeamIdFromKey } from "./team";
+
+/**
+ * An object containing the UUID of an issue or an object containing the issue key.
+ * The issue key is formatted like `LNR-123`
+ */
+export type IssueSelection = { id: string } | { key: string };
+
+export const IssueDetails = gql`
+  fragment IssueDetails on Issue {
+    id
+    title
+    number
+    description
+    priority
+    estimate
+    assignee
+    labels {
+      id
+      name
+    }
+  }
+`;
+
+export const getAllIssues = async (client: Linear) =>
+  client.request(gql`
+    ${IssueDetails}
+    query GetAllIssuesFromCLI {
+      issues {
+        ...IssueDetails
+      }
+    }
+  `);
+
+export const getIssue = async (client: Linear, selection: IssueSelection) => {
+  if ("id" in selection) {
```

Comment:
This is an example of a selection in practice. An issue can be selected by `id` or `key`. This infrastructure wouldn't be necessary if and issue was queryable by both of these fields. 

Granted, GraphQL doesn't support input unions so there's no way in the schema to express that either `id` _or_ `key` is required. So very unfortunate...

Reply:
I think we could expand the API to support queries where one passes the key ID (XXX-123) as `id`, and same for teams as well. Then creating issues via the API would be much more simple

Reply:
Yep!